qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -42,26 +48,23 @@ def __init__(self, circuit: QuantumCircuit,
             circuit: The (parametrized) quantum circuit that generates the samples of this network.
             input_params: The parameters of the circuit corresponding to the input.
             weight_params: The parameters of the circuit corresponding to the trainable weights.
-            interpret: Determines the output format, possible choices are:
-                * 'tuple' (default): a tuple of binary values, e.g. (0, 1, 0, 1, 0)
-                * 'str': a bitstring of type str, e.g. '01010'
-                * 'int': an integer corresponding to the bitstring, e.g. 10
-                * a custom callable that takes a sample of type 'tuple' and maps it to some other
-                output, output should be hashable for sparse representation of probabilities
-                and probability gradients.
-            dense: Whether to return a dense (array with 'output_shape') or sparse (dict)
-                probabilities. Dense probabilities require "interpret == 'int'" where the integer
-                will be the index in the array of probabilities.
-                TODO: what about "return_samples"??? (cf. base class)
-                TODO: update return types to handle dictionaries and arrays
-            output_shape: Gives the output_shape in case of a custom interpret callable. If this is
-                None, the output_shape is set to 1.
+            dense: Returns whether the output is dense or not.
+            return_samples: Determines whether the network returns a batch of samples or (possibly
+                sparse) array of probabilities in its forward pass. In case of probabilities,
+                the backward pass returns the probability gradients, while it returns (None, None)
+                in the case of samples. Note that return_samples==True will always result in a
+                dense return array independent of the other settings.
+            interpret: A callable that maps the measured integer to another unsigned integer or
+                tuple of unsigned integers. These are used as new indices for the (potentially
+                sparse) output array. If this is used, the output shape of the output needs to be
+                given as a separate argument.
+            output_shape: The output shape of the custom interpretation. The output shape is
+                automatically determined in case of return_samples==True.
             gradient: The gradient converter to be used for the probability gradients.
             quantum_instance: The quantum instance to evaluate the circuits.
 
         Raises:
-            QiskitMachineLearningError: if an incorrect value for `interpret` or `output_shape`
-                is passed.
+            QiskitMachineLearningError: if `interpret` is passed without `output_shape`.
         """
 
         # TODO: currently cannot handle statevector simulator, at least throw exception
```

Comment:
I see in the _sample method below a test for statevector - does this mean this TODO is outdated and can be removed. Should the test be done earlier, so it can be caught earlier rather than later, by that I mean check when the QI is provided or is there other function where this is still useful.

I did not see any setter for QI like we often have to allow it to be changed.

Reply:
removed TODO and added QI setter.